collector/plugins/sinks/wavefront/sink.go

```diff
@@ -196,6 +196,12 @@ func (sink *wavefrontSink) Export(batch *metrics.Batch) {
 		if point == nil {
 			continue
 		}
+
+		// US1004011: Source to match nodename if it exists
+		if newSource, ok := point.Tags()[metrics.LabelNodename.Key]; ok {
```

Comment:
is there any chance of Tags() returning nil or LabelNodename itself?

Reply:
If Tags() did return nil, it would just be treated as an empty map here. As far as I can tell, metrics.LabelNodename.Key always resolves to the string "nodename" so this code snippet should be safe.